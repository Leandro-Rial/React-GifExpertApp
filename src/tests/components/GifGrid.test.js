import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Prueba GifGrid", () => {
  const category = "Dragon Ball";

  // const wrapper = shallow(<GifGrid category={category} />)

  test("debe de obtener <GifGrid /> correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/cualquier/cosa.png",
        title: "Cualquier Cosa",
      },
      {
        id: "123",
        url: "http://localhost/cualquier/cosa123.png",
        title: "Cualquier Cosa 123",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
