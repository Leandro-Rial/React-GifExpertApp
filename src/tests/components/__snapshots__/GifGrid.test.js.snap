// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba GifGrid debe de obtener <GifGrid /> correctamente 1`] = `
<div
  className="card-grid-wrapper"
>
  <h3>
    Dragon Ball
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
